python/mxnet/gluon/trainer.py

```diff
@@ -65,7 +74,10 @@ def __init__(self, params, optimizer, optimizer_params=None, kvstore='device'):
                     "First argument must be a list or dict of Parameters, " \
                     "got list of %s."%(type(param)))
             self._params.append(param)
-
+        if compress_params:
+            if not isinstance(compress_params, dict):
+                raise ValueError("compress_params needs to be a dictionary")
```

Comment:
why does it have to be a dict?

Reply:
I was thinking different forms of compression might need different parameters so made it a dict. I'll change it to  kwargs style, so we don't need a dict

Reply:
Ok then please leave it as a dict here, similar to optimizer params
call it compression_params